### DISCUSSION TITLE
How to retrieve all contacts that have a phone number?

### DISCUSSION THREAD

**Discussion by behelit** [vestrel00/contacts-android#267] [Q&A] [answered]
This should be a common query as a lot of call based apps would require it.
I could not work out a method to obtain all contacts that have a phone number.
Is this possible with the current API's?

**Comment by vestrel00** [comment, marked as answer]
Hey @behelit! Thanks for asking this question! I converted your issue https://github.com/vestrel00/contacts-android/issues/266, to this Q&A discussion, so that other people can see it too 😁 

This is already possible with the current APIs provided in this library. You can take a look at the cheatsheet; https://vestrel00.github.io/contacts-android/cheatsheet/#query-contacts-advanced

<img width="696" alt="Screen Shot 2022-10-09 at 8 47 39 PM" src="https://user-images.githubusercontent.com/4332753/194811064-bcf883ea-c792-4df8-a985-8ca6cb68bb27.png">

The cheatsheet contains more code samples that you might find useful. Code samples are available in both Kotlin (as seen above) or Java (as you can see below).

<img width="658" alt="Screen Shot 2022-10-09 at 8 49 00 PM" src="https://user-images.githubusercontent.com/4332753/194811294-ecb2d5ff-8809-4294-a196-6908e275cd27.png">

If this answered your question, please mark this as the answer 🙏

**Reply by vestrel00** [reply]
It's been almost a month. I'll mark this as answer. Feel free to comment if it did not answer your question. Thanks!
